src/components/ConditionalFilter/useContainerState.ts

```diff
@@ -1,11 +1,23 @@
-import { useEffect, useState } from "react";
+import { useEffect, useRef, useState } from "react";
 
 import { FilterContainer, FilterElement } from "./FilterElement";
 import { FilterValueProvider } from "./FilterValueProvider";
 
 type StateCallback = (el: FilterElement) => void;
 type Element = FilterContainer[number];
 
+export interface UseContainerStateOptions {
+  /**
+   * Only sync with valueProvider on first mount, not on subsequent changes.
+   *
+   * Use this for modals where `valueProvider.value` is wrapped/computed
+   * (e.g., with injected constraints). The wrapping creates new array
+   * references on each render, causing unwanted resyncs that overwrite
+   * user's unsaved edits.
+   */
+  syncOnce?: boolean;
```

Comment:
this is bandaid for poor design - we should refactor our state management depending on URLs into useSyncExternalStore in React 18+